lisp/core/install-file.el

```diff
@@ -0,0 +1,55 @@
+;;; core/install-file.el --- Install packages from files, .tar files, or directories  -*- lexical-binding: t; -*-
+
+;;; Commentary:
+;;
+;; Command use to install packages from files, .tar files, or directories
+;;
+;;   $ eask install-file [files..]
+;;
+;;
+;;  Positionals:
+;;
+;;    [files..]     files to install as packages; it will install through the
+;;                  function `package-install-file'
+;;
+
+;;; Code:
+
+(let ((dir (file-name-directory (nth 1 (member "-scriptload" command-line-args)))))
+  (load (expand-file-name "_prepare.el"
+                          (locate-dominating-file dir "_prepare.el"))
+        nil t))
+
+(defun eask-install-file--guess-name (file)
+  "Guess the package name of the install FILE."
+  (file-name-sans-extension (file-name-nondirectory (directory-file-name file))))
+
+(defun eask-install-file--packages (files)
+  "The file install packages with FILES."
+  (let* ((deps (mapcar (lambda (file)
+                         (list (eask-install-file--guess-name file) file))
+                       files))
+         (names (mapcar #'car deps))
+         (len (length deps))
+         (s (eask--sinr len "" "s"))
+         (pkg-not-installed (cl-remove-if #'package-installed-p names))
+         (installed (length pkg-not-installed)) (skipped (- len installed)))
+    (eask-log "Installing %s specified file package%s..." len s)
+    (eask-msg "")
+    (eask--package-mapc (lambda (dep &rest _)
+                          (apply #'eask-package-install-file dep))
```

Comment:
The --force flag is checked in `eask-package-install-file` but installed packages are filtered before calling it. I think this means that `--force` has no effect.

Reply:
> The --force flag is checked in eask-package-install-file but installed packages are filtered before calling it. I think this means that --force has no effect.

No, those are used for calculating installed packages. It's not the prettiest solution. 😅

Reply:
Ah it's used later in a different place.